### PR TITLE
Enable ResolverStyle.STRICT for java formatters

### PR DESCRIPTION
**Pull request by pgomulka**
Joda was using ResolverStyle.STRICT when parsing. This means that date will be validated to be a correct year, year-of-month, day-of-month
However, we also want to make it works with Year-Of-Era as Joda used to, hence custom temporalquery.localdate in `DateFormatters.from`
Within `DateFormatters` we use the correct `uuuu` year instead of `yyyy` year of era

worth noting: if `yyyy`(without an era) is used in code, the parsing result will be a `TemporalAccessor` which will fail to be converted into LocalDate. We mostly use `DateFormatters.from` so this takes care of this. If possible the `uuuu` format should be used.

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Comment by pgomulka** [comment]
Ok to test

**Comment by pgomulka** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/common/time/JavaDateMathParser.java [hunk 1]
```diff
@@ -48,14 +47,14 @@
 public class JavaDateMathParser implements DateMathParser {
 
     private final JavaDateFormatter formatter;
-    private final DateTimeFormatter roundUpFormatter;
     private final String format;
+    private JavaDateFormatter roundupParser2;
 
-    JavaDateMathParser(String format, JavaDateFormatter formatter, DateTimeFormatter roundUpFormatter) {
+    JavaDateMathParser(String format, JavaDateFormatter formatter, JavaDateFormatter roundupParser2) {
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/common/time/JavaDateMathParser.java]: I think the old name was better?

**Review by rjernst — COMMENTED** [review]
One small comment, but my larger concern is adding leniency for `yyyy`. I think we should fix our tests, and enforce the deprecation we have already made for `yyyy`, where we suggested to users changing to `uuuu`.

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Comment by pgomulka** [comment]
ok to test

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by pgomulka** [comment]
> One small comment, but my larger concern is adding leniency for yyyy. I think we should fix our tests, and enforce the deprecation we have already made for yyyy, where we suggested to users changing to uuuu.
@rjernst I am fully supportive of this. I wonder if this should not be in a separate PR thought. there is >2000 usages of `yyyy` in our code base. A lot of this is in docs (which is part of the testing anyway). Shouldn't be too hard to fix, yet it would hide the purpose of this PR.
wdyt?

**Comment by rjernst** [comment]
Sure, a separate PR (or multiple) is fine.

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**